File: frontend/src/utils/debug.ts
import { currentConfig } from '../config'

export const debug = {
  log: (...args: any[]) => {
    if (currentConfig.debug) {
      console.log(...args)
    }
  },
  
  warn: (...args: any[]) => {
    if (currentConfig.debug) {
      console.warn(...args)
    }
  },
  
  error: (...args: any[]) => {
    if (currentConfig.debug) {
      console.error(...args)
    }
  }
} 

File: backend/app/utils/debug.py
from ..config import current_config


class Debug:
    def __init__(self):
        self.enabled = current_config["debug"]

    def log(self, *args):
        if self.enabled:
            print("[DEBUG]", *args)

    def error(self, *args):
        if self.enabled:
            print("[ERROR]", *args)

    def warn(self, *args):
        if self.enabled:
            print("[WARN]", *args)


debug = Debug()
